devel/README.md

```diff
@@ -0,0 +1,54 @@
+# Developing on Istio
+
+See also the [broker](https://github.com/istio/broker/blob/master/doc/dev/development.md) and [mixer](https://github.com/istio/mixer/blob/master/doc/dev/development.md)
```

Comment:
These two files are nearly identical. I just cut & pasted them last week. I think a lot of their content could be moved here and slightly generalized so it applies to all repos. They talk about the GitHub flow for example, which should be the same for all our repos. So I think we should pull common info out of these two docs and pour into here. We can do this in a separate PR or this one, but I think that should be the goal here...